src/display/editor/signature.js

```diff
@@ -131,21 +131,29 @@ class SignatureEditor extends DrawingEditor {
     input.click();
 
     const bitmap = await promise;
-    if (!bitmap?.bitmap) {
-      this.remove();
-      return;
-    }
     const {
       rawDims: { pageWidth, pageHeight },
       rotation,
     } = this.parent.viewport;
-    const drawOutlines = SignatureExtractor.process(
-      bitmap.bitmap,
-      pageWidth,
-      pageHeight,
-      rotation,
-      SignatureEditor._INNER_MARGIN
-    );
+    let drawOutlines;
+    if (bitmap?.bitmap) {
+      drawOutlines = SignatureExtractor.process(
+        bitmap.bitmap,
+        pageWidth,
+        pageHeight,
+        rotation,
+        SignatureEditor._INNER_MARGIN
+      );
+    } else {
+      drawOutlines = SignatureExtractor.extractContoursFromText(
+        "Hello PDF.js' World !!",
```

Comment:
Left-over debug statement?

Reply:
No it's just to test that the rendering is correct.
If you want to see the result, you can add a signature and cancel the file picker.
Anyway, it'll be removed in a following patch with the new UI: I'm currently splitting by my big patch into smaller ones.